### ISSUE TITLE
Bourbon support

### ISSUE THREAD

**Issue by hetsch** [bdkjones/CodeKit-1-Old#187]
Hi,

first thank you for your really helpful piece of code. Do you think it might be possible make Codekit playing nicely with [Bourbon](https://github.com/thoughtbot/bourbon)?

The problem is that bourbon uses sass extensions and i have to load them before compiling.
https://github.com/lynnwallenstein/My-Boilerplate/tree/master/css/sass/bourbon/lib

Thanks a lot...

**Comment by pushred** [comment]
Would love to see this. Compass is so bloated and duplicates so much of CodeKit.. Bourbon is a much tighter collection of mixins that make working with CSS3 so much easier.

**Comment by hetsch** [comment]
Can I help to resolve this ticket?

My problem is that i don't know how to register the sass extension.
I know that config.rb get's called by the compass plugin. Would this also be possible to register the sass extension needed by fbourbon?

**Comment by doginthehat** [comment]
+1 on this. 

That would be awesome to be able to use bourbon and codekit together..

The bourbon doc directs to add the extra library using sass' -r parameter (as explained under [Rails 2.3 or a non-Rails project](https://github.com/thoughtbot/bourbon).

Maybe there would just need a way to specify an additional library to pass to sass for compilation (could be a global pref, a per project pref or even an individual file preference)

**Comment by bdkjones** [comment]
I'm not a Compass expert, but I believe the -r flag for the Compass command line executable does exactly the same thing as adding a "require" statement to your project's Compass configuration file. 

See http://incident57.com/codekit/help.php

On that page, about halfway down, there's a section that talks about Compass and shows an example of how you would add the Zurb Foundation extension to a Compass project in CodeKit. (Basically, you just add a require statement to your configuration file that has the full, root-relative path to the installed gem.)

Do this for Bourbon's gem and see what happens. Hopefully, it will just magically work.

**Comment by doginthehat** [comment]
@bdkjones Good thinking but that doesn't work.

To use the config.rb file you have to run the codekit project as a Compass project (CodeKit actually automatically turns the projects into a Compass one as soon as the config.rb file is added - it's one of the prefs I think)..

Anyway, that doesn't work; without running it as a Compass project, the config file seems to be ignored.

**Comment by doginthehat** [comment]
Also, seems ironically pointless to have to compile with Compass to use Bourbon :)

**Comment by bdkjones** [comment]
Compass projects always compile with Compass. Make sure the project is set to use Compass and that the compass icon appears next to the project's icon.

-Bryan

**Comment by doginthehat** [comment]
@bdkjones This is about Bourbon support. We don't want to use Compass.

**Comment by bdkjones** [comment]
You don't need to use Compass for actual content. You simply add the project, then choose "Compass > Use Compass on this project" and in the dialog that pops up, choose "create new config file". Then, add the require statement for the Bourbon gem to that config file. (with a full root-relative path.)

The go create a SASS file and use Bourbon's content with appropriate import statements in the file.

When you compile, although Compass technically runs, it doesn't generate any code -- it just calls the Bourbon gem.

This should get you up and running with Bourbon.

-Bryan

**Comment by doginthehat** [comment]
@bdkjones aaaah, nice.. thanks for your patience..

So, added `require File.join(File.dirname(__FILE__), "/bourbon/lib/bourbon")` to the top of the config.rb file and CodeKit seems to pick that ok.

But CodeKit is now throwing the following error:

```
LoadError on line 1 of /[...]/bourbon/lib/bourbon.rb: no such file to load -- bourbon/generator
Run with --trace to see the full backtrace
```

Seems it can't find the Bourbon gem. Is CodeKit running with its own set of gems or something?

**Comment by doginthehat** [comment]
I'm using rbenv, maybe that has some affect on it.. ?

**Comment by bdkjones** [comment]
Again, the path needs to be FULL, ROOT-RELATIVE. 

Do not use the placeholder. Follow the example on http://incident57.com/codekit/help.php

-Bryan

**Comment by doginthehat** [comment]
`File.join(File.dirname(__FILE__), "/bourbon/lib/bourbon")` will create a full path from a relative one. 

It doesn't work either if I hard-code the full path anyway. 

The message is not that it can't load the `bourbon.rb`, the message is that _inside bourbon.rb_ it can't load 'bourbon/generator' (which is the first line inside bourbon.rb: `require "bourbon/generator"` )

So it looks like it's not picking the bourbon gem basically.

Any way I can force compass to use the bourbon gem as well?

**Comment by bdkjones** [comment]
Try adding Bourbon as a CodeKit Framework.

I've just heard from someone who did this and he claims it works great. Bourbon is, essentially, a bunch of SCSS files. Add /bourbon/app/assets/stylesheets as a CodeKit Framework and see what happens.

**Comment by doginthehat** [comment]
@bdkjones Good thinking.. 

Can't get it to work though.. once compile my css still contains untranslated bourbon functions..

E.g.

`@include box-shadow(0 2px 1px 1px rgba(#222, 0.05));` is outputted as:

```
 -webkit-box-shadow: compact(0 2px 1px 1px rgba(34, 34, 34, 0.05), false, false, false, false, false, false, false, false);
  -moz-box-shadow: compact(0 2px 1px 1px rgba(34, 34, 34, 0.05), false, false, false, false, false, false, false, false);
  -ms-box-shadow: compact(0 2px 1px 1px rgba(34, 34, 34, 0.05), false, false, false, false, false, false, false, false);
  -o-box-shadow: compact(0 2px 1px 1px rgba(34, 34, 34, 0.05), false, false, false, false, false, false, false, false);
  box-shadow: compact(0 2px 1px 1px rgba(34, 34, 34, 0.05), false, false, false, false, false, false, false, false);
```

**Comment by stewart** [comment]
The reason you can't just add bourbon as a framework is that it shells out to Ruby for a few functions, including the `compact` function you see in @doginthehat's code. Either the `bourbon` gem must be required, or the `bourbon/lib/bourbon.rb` file has to be loaded properly for the framework to function.

So the syntax for a regular SASS project would be something like:

```
sass --watch scss:css -r ./bourbon/lib/bourbon.rb
```

**Comment by doginthehat** [comment]
@stewart Yes, that's how I run bourbon with Sass.

Is there any way to tell codekit to load that rb file as well?

**Comment by paulmist** [comment]
Are we getting close to a solution/implementation on this? Anything I can do to help?

**Comment by AaronHolbrook** [comment]
Really hoping this gets resolved - I'd really like to use Bourbon with Codekit.

Having the same issue @doginthehat has had, when loaded from the config.rb file with hardcoded path, it can't find the `require "bourbon/generator"`.

Tried adding Bourbon as framework, but again had the same issue @doginthehat had, which was that it didn't run the functions as @stewart mentioned.

So @bdkjones, is there any way to tell codekit to load the bourbon.rb file correctly so that we may use this with Codekit?

**Comment by bdkjones** [comment]
Simple. Install Compass at the command line. (Make sure you put it in the normal location; do not use RVM). 

Then, go to the preferences window, select the Compass language, and tell CodeKit to use the external compiler that you just installed. You will no longer need root-relative require statements in the config file and everything should just work.

**Comment by kerns** [comment]
@bdkjones A little bourbon [told me](https://github.com/thoughtbot/bourbon/issues/134#issuecomment-10032155) that CodeKit might soon be an option for us.

**Comment by bdkjones** [comment]
Indeed. There's an update shipping in the next couple of days, but that one will not include Bourbon. The next update will and should be available in a couple weeks.

**Comment by bdkjones** [comment]
Implemented in 1.3.5, out now. Thanks!
